sdk/search/Azure.Search.Documents/src/Utilities/SearchExtensions.cs

```diff
@@ -69,5 +70,44 @@ public static string GetSearchServiceName(this Uri endpoint)
                 new List<string>() :
                 // TODO: #10600 - Verify we don't need to worry about escaping
                 new List<string>(value.Split(','));
+
+        /// <summary>
+        /// Joins a collection of items using <see cref="object.ToString"/> with the given <paramref name="separator"/>.
+        /// </summary>
+        /// <typeparam name="T">The non-nullable type of item to join.</typeparam>
+        /// <param name="items">The collection of items to join.</param>
+        /// <param name="separator">The separator between items to join. The default is a comma.</param>
+        /// <returns>
+        /// A collection of items using <see cref="object.ToString"/> with the given <paramref name="separator"/>,
+        /// or null if <paramref name="items"/> is null or empty, or contains only null or empty values.
+        /// </returns>
+        public static string JoinAsString<T>(this IEnumerable<T> items, char separator = ',')
```

Comment:
Why not String.Join?

Reply:
Because I'm not joining strings. I'm joining `T.ToString`. I could do this with a `Select` method, but those are more expensive at runtime and gains me nothing than avoiding a few extra lines of code. But, yes, at the same time I also wanted to avoid serializing empty values in the generic case. The behavior is also slightly different than what Ted had, or I'd have modified that to take an optional separator.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.string.join?view=netframework-4.8#System_String_Join__1_System_String_System_Collections_Generic_IEnumerable___0__ ?

Reply:
Thanks. I forgot about that one. But without the type constraint, it boxes `T`. Maybe in practice this happens little enough that any benefit from having an NGEN'd Framework would provide for this method, but in Core it should matter less. I'll run some quick benchmarks.

Reply:
I'm allocating more over a thousand iterations, so I'll use the suggestion.

|       Method |           Job |       Runtime |     Mean |   Error |   StdDev |   Median | Ratio | RatioSD |   Gen 0 |Gen 1 | Gen 2 | Allocated |
|------------- |-------------- |-------------- |---------:|--------:|---------:|---------:|------:|--------:|--------:|------:|------:|----------:|
|   StringJoin |    .NET 4.7.2 |    .NET 4.7.2 | 116.1 us | 2.28 us |  4.61 us | 115.6 us |  0.98 |    0.05 | 24.7803 |    - |     - | 101.86 KB |
| JoinAsString |    .NET 4.7.2 |    .NET 4.7.2 | 157.2 us | 4.87 us | 13.73 us | 156.0 us |  1.39 |    0.13 | 49.5605 |    - |     - | 203.72 KB |
|   StringJoin | .NET Core 2.1 | .NET Core 2.1 | 118.5 us | 2.33 us |  4.91 us | 118.3 us |  1.00 |    0.00 | 24.7803 |    - |     - | 101.56 KB |
| JoinAsString | .NET Core 2.1 | .NET Core 2.1 | 151.1 us | 6.79 us | 19.81 us | 147.5 us |  1.33 |    0.17 | 49.5605 |    - |     - | 203.13 KB |
|   StringJoin | .NET Core 3.1 | .NET Core 3.1 | 120.0 us | 4.57 us | 13.33 us | 118.7 us |  1.00 |    0.12 | 24.7803 |    - |     - | 101.56 KB |
| JoinAsString | .NET Core 3.1 | .NET Core 3.1 | 129.0 us | 6.62 us | 19.32 us | 119.2 us |  1.14 |    0.15 | 49.5605 |    - |     - | 203.13 KB |